libraries/3rdparty/mbedtls_utils/mbedtls_error.c

```diff
@@ -0,0 +1,1583 @@
+#endif
+
+#if defined( MBEDTLS_CHACHA20_C )
+    #include "mbedtls/chacha20.h"
+#endif
+
+#if defined( MBEDTLS_CHACHAPOLY_C )
+    #include "mbedtls/chachapoly.h"
+#endif
+
+#if defined( MBEDTLS_CIPHER_C )
+    #include "mbedtls/cipher.h"
+#endif
+
+#if defined( MBEDTLS_CMAC_C )
+    #include "mbedtls/cmac.h"
+#endif
+
+#if defined( MBEDTLS_CTR_DRBG_C )
+    #include "mbedtls/ctr_drbg.h"
+#endif
+
+#if defined( MBEDTLS_DES_C )
+    #include "mbedtls/des.h"
+#endif
+
+#if defined( MBEDTLS_DHM_C )
+    #include "mbedtls/dhm.h"
+#endif
+
+#if defined( MBEDTLS_ECP_C )
+    #include "mbedtls/ecp.h"
+#endif
+
+#if defined( MBEDTLS_ENTROPY_C )
+    #include "mbedtls/entropy.h"
+#endif
+
+#if defined( MBEDTLS_GCM_C )
+    #include "mbedtls/gcm.h"
+#endif
+
+#if defined( MBEDTLS_HKDF_C )
+    #include "mbedtls/hkdf.h"
+#endif
+
+#if defined( MBEDTLS_HMAC_DRBG_C )
+    #include "mbedtls/hmac_drbg.h"
+#endif
+
+#if defined( MBEDTLS_MD_C )
+    #include "mbedtls/md.h"
+#endif
+
+#if defined( MBEDTLS_MD2_C )
+    #include "mbedtls/md2.h"
+#endif
+
+#if defined( MBEDTLS_MD4_C )
+    #include "mbedtls/md4.h"
+#endif
+
+#if defined( MBEDTLS_MD5_C )
+    #include "mbedtls/md5.h"
+#endif
+
+#if defined( MBEDTLS_NET_C )
+    #include "mbedtls/net_sockets.h"
+#endif
+
+#if defined( MBEDTLS_OID_C )
+    #include "mbedtls/oid.h"
+#endif
+
+#if defined( MBEDTLS_PADLOCK_C )
+    #include "mbedtls/padlock.h"
+#endif
+
+#if defined( MBEDTLS_PEM_PARSE_C ) || defined( MBEDTLS_PEM_WRITE_C )
+    #include "mbedtls/pem.h"
+#endif
+
+#if defined( MBEDTLS_PK_C )
+    #include "mbedtls/pk.h"
+#endif
+
+#if defined( MBEDTLS_PKCS12_C )
+    #include "mbedtls/pkcs12.h"
+#endif
+
+#if defined( MBEDTLS_PKCS5_C )
+    #include "mbedtls/pkcs5.h"
+#endif
+
+#if defined( MBEDTLS_PLATFORM_C )
+    #include "mbedtls/platform.h"
+#endif
+
+#if defined( MBEDTLS_POLY1305_C )
+    #include "mbedtls/poly1305.h"
+#endif
+
+#if defined( MBEDTLS_RIPEMD160_C )
+    #include "mbedtls/ripemd160.h"
+#endif
+
+#if defined( MBEDTLS_RSA_C )
+    #include "mbedtls/rsa.h"
+#endif
+
+#if defined( MBEDTLS_SHA1_C )
+    #include "mbedtls/sha1.h"
+#endif
+
+#if defined( MBEDTLS_SHA256_C )
+    #include "mbedtls/sha256.h"
+#endif
+
+#if defined( MBEDTLS_SHA512_C )
+    #include "mbedtls/sha512.h"
+#endif
+
+#if defined( MBEDTLS_SSL_TLS_C )
+    #include "mbedtls/ssl.h"
+#endif
+
+#if defined( MBEDTLS_THREADING_C )
+    #include "mbedtls/threading.h"
+#endif
+
+#if defined( MBEDTLS_X509_USE_C ) || defined( MBEDTLS_X509_CREATE_C )
+    #include "mbedtls/x509.h"
+#endif
+
+#if defined( MBEDTLS_XTEA_C )
+    #include "mbedtls/xtea.h"
+#endif
+
+
+const char * mbedtls_strerror_highlevel( int errnum )
+{
+    const char * rc = NULL;
+    int use_ret;
+
+    if( errnum < 0 )
+    {
+        errnum = -errnum;
+    }
+
+    if( errnum & 0xFF80 )
+    {
+        use_ret = errnum & 0xFF80;
+
+        /* High level error codes */
+        /* */
+        /* BEGIN generated code */
+        #if defined( MBEDTLS_CIPHER_C )
```

Comment:
Use of ```switch``` would result in a faster code, right? (The same was suggested by ARM here: https://github.com/ARMmbed/mbedtls/pull/3176).

Reply:
Will look into tackling this in a separate PR to avoid scope-creep.